src/script/lua_api/l_mainmenu.cpp

```diff
@@ -637,17 +637,30 @@ int ModApiMainMenu::l_create_world(lua_State *L)
 	const char *name	= luaL_checkstring(L, 1);
 	int gameidx			= luaL_checkinteger(L,2) -1;
 
+	std::string new_dir = name;
+	fs::FixBadFilename(new_dir);
 	std::string path = porting::path_user + DIR_DELIM
-			"worlds" + DIR_DELIM
-			+ name;
+			+ "worlds" + DIR_DELIM + new_dir;
+
+	if (fs::PathExists(path)){
+		path = path + "_";
+		for (int i = 1; i <= 10000; i++){
+			std::ostringstream test_path;
+			test_path << path << i;
+			if (!fs::PathExists(test_path.str())){
+				path = test_path.str();
+				break;
+			}
+		}
+	}
 
 	std::vector<SubgameSpec> games = getAvailableGames();
 
 	if ((gameidx >= 0) &&
```

Comment:
please may you remove the superfluous parentheses here given that you touch the lines around it